docs/source/model_design/programming_vivarium/overview_engineering_files/index.rst

```diff
@@ -29,12 +29,159 @@ Overview of Engineering Files
 
 .. contents::
    :local:
-   :depth: 1
+   :depth: 2
 
-.. todo::
 
-  Add to this page. Include explanations for the relevant engineering files (model specs, branches file, loader.py, etc.) that includes when the functions are called, when you might need to look at them, what information is stored where, and any other relevant information for RT based tasks. 
+This document is designed to provide an overview of the files found in the 
+engineering repository and some examples of when you might need to use 
+each file. This is NOT inclusive of files as some are less relevant 
+for research team members, and the files included change between models. 
 
-  Integrate links to: `YAML basics <https://vivarium.readthedocs.io/en/latest/concepts/model_specification/yaml_basics.html#model-specification-yaml-concept>`_, `model specs file <https://vivarium.readthedocs.io/en/latest/concepts/model_specification/index.html>`_, and the `branches file <https://vivarium-cluster-tools.readthedocs.io/en/latest/branch.html>`_. 
+This page is organized in sections by the different folders found in engineering 
+repositories and then the pages within folders, as relevant. 
 
+Components
+----------
 
+The :code:`components` folder contains files on the information for each 
+modeling "component". This includes things like disease models, 
+risk factors, pregnancies, the health care system, or 
+interventions. 
+
+The files included will be specific to your project rather than generalizable. 
+Additionally, these files will mostly be maintained and written by the 
+engineering team. For those reasons, we won't elaborate on these files here. 
+
+Constants
+---------
+
+The :code:`constants` folder contains input information for the model. This includes 
+things like data values, locations, file paths, draw counts, column 
+names, and outputs needed. 
+
+There are a few reasons you might use files from this folder - changing data inputs, 
+adjusting information for the artifact, or validating file paths. The below files 
+are not all inclusive, but represent most of what you should be aware of. 
+
+data_keys
++++++++++
+
+The :code:`data_keys.py` file dictates which keys are run to make the artifact. Usually, the 
+list of data keys is at the bottom of the file. More information on how and when to 
+edit this can be found on the :ref:`artifact building page <artifact_building_rt>`. 
+
+metadata
+++++++++
+
+The :code:`metadata.py` file will mainly need to be edited to change the locations for artifact 
```

Comment:
It might be obvious, but should we say that this a file for storing metadata related constants in addition to the location information? The locations in this file also define which locations we _can_ make artifacts for. We specify which locations we'd like to make artifacts for from the command line, but artifact building will fail if we specify a location that isn't contained in this file.